pkg/providers/version/version.go

```diff
@@ -48,25 +55,44 @@ type DefaultProvider struct {
 	cache               *cache.Cache
 	cm                  *pretty.ChangeMonitor
 	kubernetesInterface kubernetes.Interface
+	eksapi              eksiface.EKSAPI
 }
 
-func NewDefaultProvider(kubernetesInterface kubernetes.Interface, cache *cache.Cache) *DefaultProvider {
+func NewDefaultProvider(kubernetesInterface kubernetes.Interface, cache *cache.Cache, eksapi eksiface.EKSAPI) *DefaultProvider {
 	return &DefaultProvider{
 		cm:                  pretty.NewChangeMonitor(),
 		cache:               cache,
 		kubernetesInterface: kubernetesInterface,
+		eksapi:              eksapi,
 	}
 }
 
 func (p *DefaultProvider) Get(ctx context.Context) (string, error) {
+	var version string
 	if version, ok := p.cache.Get(kubernetesVersionCacheKey); ok {
 		return version.(string), nil
 	}
-	serverVersion, err := p.kubernetesInterface.Discovery().ServerVersion()
+	output, err := p.eksapi.DescribeClusterWithContext(ctx, &eks.DescribeClusterInput{
+		Name: lo.ToPtr(options.FromContext(ctx).ClusterName),
+	})
 	if err != nil {
-		return "", err
+		if !awserrors.HasNoAccess(err) {
+			return "", err
+		} else {
+			output, err := p.kubernetesInterface.Discovery().ServerVersion()
+			if err != nil {
+				return "", err
+			} else if output != nil {
+				version = fmt.Sprintf("%s.%s", output.Major, strings.TrimSuffix(output.Minor, "+"))
+				log.FromContext(ctx).Info("Successfully retrieved Kubernetes version from Kubernetes API", "version", version)
```

Comment:
```suggestion
				log.FromContext(ctx).Info("retrieved Kubernetes version from Kubernetes API", "version", version)
```

nit: we have taken a style choice to return error/info messaging starting with a lowercase letter